common/src/main/kotlin/streams/service/sink/strategy/SchemaIngestionStrategy.kt

```diff
@@ -2,38 +2,63 @@ package streams.service.sink.strategy
 
 import streams.events.*
 import streams.extensions.quote
+import streams.service.StreamsSinkEntity
 import streams.utils.IngestionUtils.getLabelsAsString
 import streams.utils.IngestionUtils.getNodeKeysAsString
-import streams.serialization.JSONUtils
-import streams.service.StreamsSinkEntity
 import streams.utils.SchemaUtils.getNodeKeys
 import streams.utils.SchemaUtils.toStreamsTransactionEvent
 import streams.utils.StreamsUtils
 
 
-private fun prepareRelationshipEvents(events: List<StreamsTransactionEvent>, withProperties: Boolean = true): Map<RelationshipSchemaMetadata, List<Map<String, Any>>> = events
-        .mapNotNull {
-            val payload = it.payload as RelationshipPayload
+class SchemaIngestionStrategy: IngestionStrategy {
 
-            if (payload.start.ids.isEmpty() || payload.end.ids.isEmpty()) {
-                null
-            } else {
-                val properties = payload.after?.properties ?: payload.before?.properties ?: emptyMap()
+    private fun prepareRelationshipEvents(events: List<StreamsTransactionEvent>, withProperties: Boolean = true): Map<RelationshipSchemaMetadata, List<Map<String, Any>>> = events
+            .mapNotNull {
+                val payload = it.payload as RelationshipPayload
 
-                val key = RelationshipSchemaMetadata(it.payload)
-                val value = if (withProperties) {
-                    mapOf("start" to payload.start.ids, "end" to payload.end.ids, "properties" to properties)
-                } else {
-                    mapOf("start" to payload.start.ids, "end" to payload.end.ids)
+                val startNodeConstraints = getNodeConstraints(it) {
+                    it.type == StreamsConstraintType.UNIQUE && payload.start.labels.orEmpty().contains(it.label)
+                }
+                val endNodeConstraints = getNodeConstraints(it) {
+                    it.type == StreamsConstraintType.UNIQUE && payload.end.labels.orEmpty().contains(it.label)
                 }
 
-                key to value
+                if (startNodeConstraints.isEmpty() || endNodeConstraints.isEmpty()) {
+                    null
+                } else {
+                    val startNodeKeys = getNodeKeys(
```

Comment:
can you extract this into a sensible method

Reply:
done!